### ISSUE TITLE
Leak reported by Netty apparently on SecureConnectionFactoryImplTest 

### ISSUE THREAD

**Issue by eolivelli** [pravega/pravega#4819]
**Problem description**
I saw this failure while running the full tests on current git sha 37614da10725e0172ff0d3b8bc2e9ea81795258a

`./gradlew clean build`

The error is not reproducible but the stacktrace gives an hint about a possible OpenSslServerContext created during the setup of SecureConnectionFactoryImplTest test but never released

```
io.pravega.client.netty.impl.SecureConnectionFactoryImplTest > getActiveChannelTestWithConnectionPooling FAILED
    java.lang.AssertionError: RESOURCE LEAK: LEAK: {}.release() was not called before it's garbage-collected. See http://netty.io/wiki/reference-counted-objects.html for more information.{}
        at org.junit.Assert.fail(Assert.java:88)
        at io.pravega.test.common.LeakDetectorTestSuite$ResourceLeakLoggerFactory$ResourceLeakAssertionLogger.error(LeakDetectorTestSuite.java:241)
        at io.pravega.test.common.LeakDetectorTestSuite$ResourceLeakLoggerFactory$ResourceLeakAssertionLogger.error(LeakDetectorTestSuite.java:235)
        at io.netty.util.ResourceLeakDetector.reportTracedLeak(ResourceLeakDetector.java:311)
        at io.netty.util.ResourceLeakDetector.reportLeak(ResourceLeakDetector.java:300)
        at io.netty.util.ResourceLeakDetector.track0(ResourceLeakDetector.java:257)
        at io.netty.util.ResourceLeakDetector.track(ResourceLeakDetector.java:245)
        at io.netty.buffer.AbstractByteBufAllocator.toLeakAwareBuffer(AbstractByteBufAllocator.java:41)
        at io.netty.buffer.PooledByteBufAllocator.newDirectBuffer(PooledByteBufAllocator.java:331)
        at io.netty.buffer.AbstractByteBufAllocator.directBuffer(AbstractByteBufAllocator.java:185)
        at io.netty.buffer.AbstractByteBufAllocator.directBuffer(AbstractByteBufAllocator.java:176)
        at io.netty.buffer.AbstractByteBufAllocator.buffer(AbstractByteBufAllocator.java:113)
        at io.netty.handler.codec.base64.Base64.encode(Base64.java:123)
        at io.netty.handler.ssl.SslUtils.toBase64(SslUtils.java:347)
        at io.netty.handler.ssl.PemX509Certificate.append(PemX509Certificate.java:128)
        at io.netty.handler.ssl.PemX509Certificate.toPEM(PemX509Certificate.java:86)
        at io.netty.handler.ssl.ReferenceCountedOpenSslContext.toBIO(ReferenceCountedOpenSslContext.java:772)
        at io.netty.handler.ssl.ReferenceCountedOpenSslServerContext.newSessionContext(ReferenceCountedOpenSslServerContext.java:172)
        at io.netty.handler.ssl.OpenSslServerContext.<init>(OpenSslServerContext.java:352)
        at io.netty.handler.ssl.OpenSslServerContext.<init>(OpenSslServerContext.java:335)
        at io.netty.handler.ssl.SslContext.newServerContextInternal(SslContext.java:422)
        at io.netty.handler.ssl.SslContextBuilder.build(SslContextBuilder.java:447)
        at io.pravega.client.netty.impl.ConnectionFactoryImplTest.setUp(ConnectionFactoryImplTest.java:64)
        at io.pravega.client.netty.impl.SecureConnectionFactoryImplTest.setUp(SecureConnectionFactoryImplTest.java:20)
```

**Problem location**
Running tests

**Suggestions for an improvement**
Check the code of SecureConnectionFactoryImplTest for resource leaks around the SslContext

**Comment by fpj** [comment]
@eolivelli will you be able to look into this?

**Comment by eolivelli** [comment]
Sure @fpj

**Comment by eolivelli** [comment]
@andreipaduroiu this looks like a false positive

the buffer pointed by the leak detector is handled safely
https://github.com/netty/netty/blob/4.1/handler/src/main/java/io/netty/handler/ssl/PemX509Certificate.java#L128
(the pointer is on current dev version of Netty but that code didn't change recently)

Btw in the test suite on Pravega we are not disposing the SslContext and other resources.
I am sending a PR about cleaning up the test class

**Comment by eolivelli** [comment]
This is the PR #4831

**Comment by RaulGracia** [comment]
Closed via #4831.
